### PR TITLE
test: Test Coverage for ERPNext Purchase Receipt Trends – Filters, Edge Cases, and Chart Data

### PR DESCRIPTION
**Pull request by hrishikesh8848**
### Title: 
Add Test Cases for Purchase Receipt Trends Report

### Test Summary:
This test suite adds automated test cases for the Purchase Receipt Trends report in ERPNext. The purpose is to ensure the report behaves correctly under different conditions, including valid inputs, empty datasets, missing filters, and chart edge cases. It covers both the execute() report logic and get_chart_data() function responsible for preparing chart configurations.

### Scenarios Covered
Input Validations
test_execute_with_valid_filters_T_PRT_001
Validates that when all required filters (company, fiscal_year, based_on, group_by, etc.) are provided, the report returns a list of columns, non-empty data, and a properly formatted chart dictionary including datasets and labels.

test_execute_with_empty_data_T_PRT_002
Simulates a scenario with a fiscal year that has no transactions and confirms the report handles it gracefully by returning an empty dataset and chart.

Edge Cases
test_chart_data_top_10_cutoff_T_PRT_003
Ensures the report chart only includes up to 10 items (suppliers/items/months) even if more data is available, keeping chart rendering performant and readable.

test_chart_data_without_group_by_T_PRT_004
Confirms that even when the group_by filter is not set, the report executes without errors and still limits chart labels to 10 items.

test_chart_data_with_no_data_T_PRT_005
Validates that get_chart_data() handles completely empty report data correctly by returning an empty or safe chart object (empty dict or list).

API Response Handling
test_execute_with_valid_filters_T_PRT_001
Verifies that the tuple returned by execute() includes:

test_chart_data_top_10_cutoff_T_PRT_003
Confirms the structure of chart["data"] adheres to expectations, especially ensuring consistent frontend compatibility with top-10 datasets.

### Technology:

Framework: FrappeTestCase (Frappe's built-in test framework)

### Linked Feature/Bug:

None

### CODE REVIEW

File: erpnext/stock/report/purchase_receipt_trends/test_purchase_receipt_trends.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+from erpnext.stock.report.purchase_receipt_trends.purchase_receipt_trends import (
+	execute,
+	get_chart_data,
+)
+
+
+class TestPurchaseReceiptTrendsReport(FrappeTestCase):
+	def setUp(self):
+		frappe.set_user("Administrator")
+		self.company = frappe.get_doc("Company", "_Test Company")
```
Comment (aasif-patel) [inline erpnext/stock/report/purchase_receipt_trends/test_purchase_receipt_trends.py:21]: How did you ensure that system have _Test Company in the database?

File: erpnext/stock/report/purchase_receipt_trends/test_purchase_receipt_trends.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+				if doc.docstatus == 1:
+					doc.cancel()
+
+		for supplier in self.supplier_names:
+			if frappe.db.exists("Supplier", supplier):
+				supplier = frappe.get_doc("Supplier", supplier)
+				supplier.disabled = 1
+
+		if frappe.db.exists("Fiscal Year", "2099-2100"):
+			frappe.delete_doc("Fiscal Year", "2099-2100", force=True)
```
Comment (aasif-patel) [inline erpnext/stock/report/purchase_receipt_trends/test_purchase_receipt_trends.py]: I think frappe.rollback will do the trick no need to select and delete the doc

**Review by aasif-patel — CHANGES_REQUESTED** [review]
all the test cases are checking datatype, or the length of the output data, None of the test checks if the data is properly executed or not,.
